Fixed as_poly() by making it raise error with unsupported data types eg: tuples

dhruvmendiratta6 opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234". See
https://github.com/blog/1506-closing-issues-via-pull-requests . Please also
write a comment on that issue linking back to this pull request once it is
open. -->
Fixes #18038 


#### Brief description of what is fixed or changed

The commit fixes Issue #18038 in which as_poly() would work with tuples and give wrong output. This was fixed by making it an attribute of Expr as tuples are not a subclass Expr. Tests for the same were added  in sympy/core/tests/test_expr.py and sympy/integrals/tests/test_integrals.py . This ultimately fixed the issue where integrate((x, x)) would give a wrong result.


#### Other comments


#### Release Notes

<!-- Write the release notes for this release below. See
https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more information
on how to write release notes. The bot will check your release notes
automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
* core
    * Moved .as_poly() from Basic to Expr.
<!-- END RELEASE NOTES -->

dhruvmendiratta6 committed:
Fixed as_poly() by making it raise error with unsupported data types eg:
tuples

The commit fixes Issue #18038 in which as_poly() would work with tuples and give wrong output. This was fixed by making it an attribute of Expr as tuples are not a subclass Expr. Tests for the same were added  in sympy/core/tests/test_expr.py and sympy/integrals/tests/test_integrals.py . This ultimately fixed the issue where integrate((x, x)) would give a wrong result.

jksuom commented on sympy/integrals/tests/test_integrals.py:
```diff
@@ -215,6 +215,10 @@ def test_issue_3560():
 
 
+def test_issue_18038():
+    raises(AttributeError, lambda: integrate((x, x))
```
One more parenthesis should be appended. Now a syntax error results.

dhruvmendiratta6 replied:
I made the changes but Travis is still showing some some error

jksuom replied:
That was an internet connection issue. I have restarted the job.

dhruvmendiratta6 committed:
Fixed a syntax error and omitted some extra blank lines left earlier

This commit fixes syntactical error left in
sympy/integrals/tests/test_integrals.py and omits extralines left in
sympy/core/basic.py .

jksuom commented:
There is one more change to be made:
```
--- a/doc/src/modules/polys/basics.rst
+++ b/doc/src/modules/polys/basics.rst
@@ -69,7 +69,7 @@ keyword parameter. By default, it is determined by the coefficients
 of the polynomial arguments.
 
 Polynomial expressions can be transformed into polynomials by the
-method :obj:`sympy.core.basic.Basic.as_poly`::
+method :obj:`sympy.core.expr.Expr.as_poly`::
 
     >>> e = (x + y)*(y - 2*z)
     >>> e.as_poly()
```

dhruvmendiratta6 committed:
Updated the documentation

jksuom commented:
Thank you for your contribution.
